sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/FeatureManager.java

```diff
@@ -73,45 +96,174 @@ public Mono<Boolean> isEnabledAsync(String feature) {
+     * @param featureContext Local context
+     * @return Assigned Variant
+     */
+    public Mono<Variant> getVariantAsync(String feature, Object featureContext) {
+        return generateVariant(feature, featureContext).single();
+    }
+
+    private Mono<Boolean> checkFeature(String feature, Object featureContext) throws FilterNotFoundException {
         if (featureManagementConfigurations.getFeatureManagement() == null
             || featureManagementConfigurations.getOnOff() == null) {
-            return false;
+            return Mono.just(false);
         }
 
         Boolean boolFeature = featureManagementConfigurations.getOnOff().get(feature);
 
         if (boolFeature != null) {
-            return boolFeature;
+            return Mono.just(boolFeature);
         }
 
         Feature featureItem = featureManagementConfigurations.getFeatureManagement().get(feature);
 
         if (featureItem == null || !featureItem.getEvaluate()) {
-            return false;
+            return Mono.just(false);
+        }
+
+        List<Mono<Boolean>> results = new ArrayList<>();
+
+        for (FeatureFilterEvaluationContext featureFilter : featureItem.getEnabledFor().values()) {
+            if (StringUtils.hasText(featureFilter.getName())) {
+                results.add(isFeatureOn(featureFilter, feature, featureContext));
+            }
         }
 
-        Stream<FeatureFilterEvaluationContext> filters = featureItem.getEnabledFor().values().stream()
-            .filter(Objects::nonNull).filter(featureFilter -> featureFilter.getName() != null);
+        if (results.size() == 0) {
+            return Mono.just(false);
+        }
 
         // All Filters must be true
         if (featureItem.getRequirementType().equals("All")) {
-            return filters.allMatch(featureFilter -> isFeatureOn(featureFilter, feature));
+            return Flux.merge(results).reduce((a, b) -> a && b).single();
         }
 
         // Any Filter must be true
-        return filters.anyMatch(featureFilter -> isFeatureOn(featureFilter, feature));
+        return Flux.merge(results).reduce((a, b) -> a || b).single();
+    }
+
+    private Mono<Variant> generateVariant(String featureName, Object featureContext) {
+
+        VariantAssignment variantAssignment = new VariantAssignment(contextAccessor, evaluationOptions, propertiesProvider);
+
+        if (!StringUtils.hasText(featureName)) {
+            throw new IllegalArgumentException("Feature Variant name can not be empty or null.");
+        }
+
+        Feature feature = featureManagementConfigurations.getFeatureManagement().get(featureName);
+
+        if (feature == null) {
+            throw new FeatureManagementException("The Feature " + featureName + " can not be found.");
+        }
+
+        validateVariant(feature, featureName);
+
+        // Disabled?
+        if (!feature.getEvaluate() && StringUtils.hasText(feature.getAllocation().getDefautlWhenDisabled())) {
```

Comment:
Seems like the `feature.getAllocation().getDefautlWhenDisabled()` is used in several places, so I suggest we extract a variable for it.